group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntime.java

```diff
@@ -541,6 +541,28 @@ private void unload() {
         }
     }
 
+    class OperationTimeout extends TimerTask {
+        private final TopicPartition tp;
+        private final DeferredEvent event;
+
+        public OperationTimeout(
+            TopicPartition tp,
+            DeferredEvent event,
+            long delayMs
+        ) {
+            super(delayMs);
+            this.event = event;
+            this.tp = tp;
+        }
+
+        @Override
+        public void run() {
+            String name = event.toString();
```

Comment:
It does not include `if (!future.isDone()) {` now. Is it ok? I don't have much context :)

Reply:
We don't need it anymore because we cancel the timer when the operation is completed. The future is completed at the same time so it is equivalent.